### PR TITLE
feat(npm)!: separate npm, pnpm, yarn managers

### LINKED ISSUE #5607: Separate npm/yarn/pnpm/lerna managers
**What would you like Renovate to be able to do?**

Have separate Renovate "managers" for npm/yarn/pnpm/lerna instead of the combined one today.

**Describe the solution you'd like**

If they were separated into different managers then it could simplify the code a lot.

**Describe alternatives you've considered**

Keeping as-is, with "npm" as a single manager.

**Additional context**

Today we support more than one type of lock file with the same `package.json`, e.g. `package-lock.json` and `yarn.lock`. This isn't a recommended config but some of our users do have it. This would mean we'd need two managers active on the same package file.

A challenge is how to know which is which, and not duplicating work:
- Yarn, pnpm and lerna managers could "abort" if they don't find their respective lock or config files
- npm could abort if it finds config files of the others or package-lock.json is in use

The above would mean that the managers all have to be "aware" of each other though, which can reduce part of the benefits of separating them.

### PR DESCRIPTION
**Pull request by rarkins**
## Context

Closes #5607

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/index.spec.ts [hunk 1]
```diff
@@ -97,6 +97,13 @@ describe('config/index', () => {
       ]);
     });
 
+    it('falls back from yarn to pnpm', () => {
```
Comment (HonkingGoose) [inline lib/config/index.spec.ts:100]: Do you really mean to have this PR title?

`feat(5607)!: separate npm, pnpm, yarn managers`

Is `(5607)` really the scope you want to use?

Comment (rarkins) [inline lib/config/index.spec.ts:100]:
```suggestion
    it('falls back from yarn to npm', () => {
```

**Comment by The-Code-Monkey** [comment]
@rarkins would it be a good idea, while your in the npm and yarn area to add #20065 in at the same time?

**Comment by rarkins** [comment]
No, that's like asking "would it be a good idea, while you're doing a heart transplant, to do a boob job at the same time?"

**Pull request closed without merging**